More info when critical failure occurs

hubertp opened the pull request:
### Pull Request Description

Log problematic module to help with debugging critical failure.
Right now it's a real guess-work.
Related to #11088.

hubertp committed:
More info when critical failure occurs

Log problematic module to help with debugging critical failure.

hubertp committed:
One more exception

JaroslavTulach commented on lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java:
```diff
@@ -131,7 +132,12 @@ public Tree parse(CharSequence input) {
+    try {
+      return Tree.deserialize(message);
+    } catch (BufferUnderflowException | IllegalArgumentException e) {
+      System.err.println("Unrecoverable parser failure for: " + input);
```
Can we use our logging framework, please?

hubertp replied:
Fair enough

hubertp committed:
s/System.err/Logger.error/

hubertp committed:
maybe append slf4j deps

hubertp committed:
fix what looks like a long standing typo

`GeneratedFormatTests.java` not `GeneratedFormatTests..java`

hubertp committed:
one more typo

hubertp committed:
Fix directory where to look for classpath

`./run java-gen test --skip-version-check` now works. At least locally.

hubertp committed:
Local is fine, CI is not. More temporary debugging...

JaroslavTulach commented:
There are still some [CI failures](https://github.com/enso-org/enso/actions/runs/10940656341/job/30373557349?pr=11092#step:7:5510).

hubertp committed:
Ensure project's managedClasspath is exported

Running java tests requires us knowing all additional dependencies as
they have to be added to the classpath manually. That can only be
ensured by invoking the right sbt target.

hubertp committed:
Move sbt call after graalvm setup

hubertp committed:
removing CI debugging

kazcw commented on build/build/src/rust/parser.rs:
```diff
@@ -63,7 +79,11 @@ pub async fn run_self_tests(repo_root: &RepoRoot) -> Result {
+        .apply(&javac::Classpath::new([
+            lib.as_path(),
+            base.as_path(),
+            Path::new(&dependencies_from_string),
```
```suggestion
            Path::new(dependencies_from_string),
```

kazcw commented on build/build/src/rust/parser.rs:
```diff
@@ -50,6 +51,21 @@ pub async fn generate_java(repo_root: &RepoRoot) -> Result {
+        .join("streams")
+        .join("export");
+
+    let path = external_dependencies_file.into_os_string();
+    let mut dependencies_from_string = fs::read_to_string(path.clone())?;
+    dependencies_from_string.pop(); // Remove newline sign
```
```suggestion
    let dependencies_from_string = fs::read_to_string(&external_dependencies_file)?;
    let dependencies_from_string = dependencies_from_string.trim_ascii_end();
```

hubertp committed:
Apply suggestions from code review

Co-authored-by: Kaz Wesley <kaz@lambdaverse.org>

JaroslavTulach commented on lib/rust/parser/generate-java/java/org/enso/syntax2/Parser.java:
```diff
@@ -131,7 +133,13 @@ public Tree parse(CharSequence input) {
+      return Tree.deserialize(message);
+    } catch (BufferUnderflowException | IllegalArgumentException e) {
+      LoggerFactory.getLogger(this.getClass())
+          .error("Unrecoverable parser failure for: {}", input, e);
```
This code is calling [error(Object,Throwable)](https://www.slf4j.org/apidocs-2.1.0/org/slf4j/Logger.html#error(java.lang.String,java.lang.Throwable)) method. What's its behavior in Enso?

#### When running `runtime-integration-tests`

Will it print the message as well as exception to the (CI) console?

#### When running in `enso` CLI mode

Will it print the message as well as exception to the (CI) console? Will it do something different?

#### When running in Enso IDE mode

What will appear in console and what will appear in associated log file?

hubertp replied:
This PR changes nothing when it comes to how log messages are consumed, this is calling a plain slf4j api. The actual printing is dependent on the implementation/configuration, as you know.

If you don't like the format/output of the logger in CLI or tests then please create a ticket for that. IDE works as expected.

JaroslavTulach replied:
> If you don't like the format/output of the logger in CLI or tests then please create a ticket for that. IDE works as expected.

Which is?

> What will appear in console and what will appear in associated log file?

Will you, please, answer my question?

JaroslavTulach reviewed (commented):
I'd like to understand what will be the behavior of the newly added `error(...)` call in various environments Enso code runs in. Thank you.
